### PR TITLE
User emails management

### PR DESCRIPTION
**Pull request by allenwq**
This PR can be reviewed commit by commit

### CODE REVIEW

File: app/controllers/user/emails_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class User::EmailsController < ApplicationController
+  load_and_authorize_resource :email, class: User::Email.name
+
+  # Set an email as the primary email
+  def set_primary
+    existing_primary_email = current_user.emails.find(&:primary?)
```
Comment (lowjoel) [inline app/controllers/user/emails_controller.rb]: Not ACID. Wrap in a transaction.

Also, it is a bad idea to have multiple resources accessed in an action.

File: app/controllers/user/emails_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    if @email.update_columns(primary: true)
+      redirect_to user_emails_path, success: 'Success'
+    else
+      redirect_to user_emails_path, danger: @email.errors
+    end
+  end
+
+  def destroy #:nodoc
+    if @email.destroy
+      redirect_to user_emails_path, success: 'Email destroyed'
```
Comment (lowjoel) [inline app/controllers/user/emails_controller.rb]: Internationalisation?

File: app/controllers/user/emails_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    else
+      redirect_to user_emails_path, error: @email.errors.message
```
Comment (lowjoel) [inline app/controllers/user/emails_controller.rb]: Does this work if there is more than one error?

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class UsersController < ApplicationController
+  before_action :load_user
```
Comment (lowjoel) [inline app/controllers/users_controller.rb]: Authorize?

File: app/controllers/user/emails_controller.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+class User::EmailsController < ApplicationController
+  load_and_authorize_resource :email, class: User::Email.name
```
Comment (lowjoel) [inline app/controllers/user/emails_controller.rb]: Through?

**Comment by lowjoel** [comment]
Is this done?

**Comment by allenwq** [comment]
@lowjoel I'll block this first, haven't got time to do this

**Comment by allenwq** [comment]
replaced by #421

**Pull request closed without merging**